lib/src/package_context.dart

```diff
@@ -132,6 +135,15 @@ class PackageContext {
     }
   }
 
+  Future<List<ScreenshotResult>> processScreenshots() async {
+    if (_screenshotResults != null) return _screenshotResults!;
+    if (pubspec.screenshots != null) {
+      _screenshotResults =
+          await processAllScreenshots(pubspec.screenshots!, packageDir);
+    }
+    return _screenshotResults!;
```

Comment:
So if `pubspec.screenshots` is not null, but an empty list, you'll get null assertion fail.
```suggestion
    if (_screenshotResults == null) {
      _screenshotResults =
          await processAllScreenshots(pubspec.screenshots ?? [], packageDir);
    }
    return _screenshotResults!;
```
or just:
```suggestion
    _screenshotResults ??=
          await processAllScreenshots(pubspec.screenshots ?? [], packageDir);
    return _screenshotResults!;
```

or just:
```suggestion
    return _screenshotResults ??=
          await processAllScreenshots(pubspec.screenshots ?? [], packageDir);
```

Reply:
Done,